src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,28 +61,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     addTermRating(request);
     response.setContentType("text/html; charset=UTF-8");
     response.setCharacterEncoding("UTF-8");
-    response.sendRedirect("/index.html");
   }
 
   public void addTermRating(HttpServletRequest request) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    String termFeedback = request.getParameter("term-input");
-    Long termRating = Long.parseLong(request.getParameter("rating-term"));
-    Long workHours = Long.parseLong(request.getParameter("hoursOfWork"));
-    Long difficulty = Long.parseLong(request.getParameter("difficulty"));
-    String professorFeedback = request.getParameter("prof-input");
-    Long professorRating = Long.parseLong(request.getParameter("rating-professor"));
+
+    StringBuilder jb = new StringBuilder();
+    String line = null;
+    try {
+      BufferedReader reader = request.getReader();
+      while ((line = reader.readLine()) != null) jb.append(line);
+    } catch (Exception exception) {
+      /*report an error*/
```

Comment:
would you now want to throw the caught exception?

Reply:
Yes! Sorry, fixed this on my other PR will implement those changes as well here.